protocols/relay/src/v2/client/transport.rs

```diff
@@ -320,12 +320,21 @@ fn parse_relayed_multiaddr(
 
 pub struct RelayListener {
     listener_id: ListenerId,
+    /// Queue of events to report when polled.
     queued_events: VecDeque<<Self as Stream>::Item>,
+    /// Channel for messages from the behaviour [`Handler`][super::handler::Handler].
```

Comment:
```suggestion
    /// Channel for messages from the behaviour [`Client`][super::Client].
```

Why reference the handler here? Wouldn't the `NetworkBehaviour` implementation be more appropriate?

Reply:
Because the NetworkBehaviour never uses the channel, it just passes it to the Handler. The handler stores it in the `Handler::reservation` and from there sends messages to the transport in `Reservation::forward_messages_to_transport_listener`. 
Took me a while myself to understand this, which is why I pointed it out in the docs.

---

Circling back to an out of band discussion we had:
The fact that the handler sends the messages instead of the behaviour was actually the reason why I decided for now to stick with the concept of Listeners in the relay transport. We currently have a 1:1 relationship between the handler of the [first connection](https://github.com/libp2p/rust-libp2p/blob/9f7d0fe0ac6bc75a9922b7e60d0d146b3e48f94a/protocols/relay/src/v2/client.rs#L245-L249) to the relay, and the transport's listener for the relayed address. They share the above `mpsc::channel::<ToListenerMsg>`, with only a single sender (that is managed by the one handler). If either side is dropped the other one is informed and can close as well, which is quite convenient.
Of course we can do without listeners and just have a mpsc channel between behaviour and transport, but I am not sure whether this would really simplify things.

Reply:
Thanks for elaborating! The above makes sense. No opinion on what is the simpler approach. Fine leaving it as is.